src/lib/parts.ts

```diff
@@ -267,15 +268,36 @@ export class NodePart implements Part {
         this.value.template === template) {
       this.value.update(value.values);
     } else {
-      // Make sure we propagate the template processor from the TemplateResult
-      // so that we use its syntax extension, etc. The template factory comes
-      // from the render function options so that it can control template
-      // caching and preprocessing.
       const instance =
           new TemplateInstance(template, value.processor, this.options);
-      const fragment = instance._clone();
-      instance.update(value.values);
-      this.__commitNode(fragment);
+
+      // This is the marker content that would have been generated by this
+      // TemplateResult.
+      const markerWithHash = `lit-part ${value.digest}`;
+      if (this.startNode.textContent === markerWithHash) {
+        // We got a match, so we're rehydrating DOM pre-rendered with this
+        // TemplateResult's Template.
+
+        // we need to give the instance the parts
+        if (this.options.prerenderedParts !== undefined) {
```

Comment:
If this is `undefined`, does that imply there are no dynamic parts inside the instance? If not, I'm not sure calling `instance.update` below will actually do anything.